File: python-leetcode/linked-list/160.相交链表.py
from typing import Optional

from type import ListNode


class Solution:
    def getIntersectionNode(
        self, headA: ListNode, headB: ListNode
    ) -> Optional[ListNode]:
        visited = set[ListNode]()

        temp = headA
        while temp:
            visited.add(temp)
            temp = temp.next

        temp = headB
        while temp:
            if temp in visited:
                return temp
            temp = temp.next

        return None

File: go-leetcode/linked-list/160.相交链表.go
package leetcode

func getIntersectionNode(headA *ListNode, headB *ListNode) *ListNode {
	visited := make(map[*ListNode]bool)

	temp := headA
	for temp != nil {
		visited[temp] = true
		temp = temp.Next
	}

	temp = headB
	for temp != nil {
		if visited[temp] {
			return temp
		}
		temp = temp.Next
	}

	return nil
}
